File: Labs/Lab4/Lab4.py
class Lab4:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def invertTree(self, root: Lab4) -> Lab4:
    if root is None:  # If no root is found then...
        return None  # return None
    # swap and invert subtrees
    root.left, root.right = self.invertTree(
        root.right), self.invertTree(root.left)
    return root  # return inverted root

File: Labs/Lab4/Lab4.java
package Labs.Lab4;

public class Lab4 {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode() {
        }

        TreeNode(int val) {
            this.val = val;
        }

        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    public TreeNode invertTree(TreeNode root) {
        // check the base case
        if (root == null) {
            return null;
        }
        // invert right subtree
        TreeNode right = invertTree(root.right);
        // invert left subtree
        TreeNode left = invertTree(root.left);
        // swap left and right
        root.left = right;
        root.right = left;
        // return the root node
        return root;
    }

}
